processor/src/main/java/com/linecorp/decaton/processor/runtime/internal/PartitionProcessor.java

```diff
@@ -143,9 +146,10 @@ private Task destroyThreadProcessorTask(int i) {
     }
 
     @Override
-    public void awaitShutdown() throws InterruptedException {
+    public void awaitShutdown(Duration limit) throws InterruptedException {
+        final Instant absLimit = Instant.now().plus(limit);
         for (ProcessorUnit unit : units) {
-            unit.awaitShutdown();
+            unit.awaitShutdown(Duration.between(Instant.now(), absLimit));
```

Comment:
According to the document it seems that the resulting Duration can be negative. Receiving negative duration is kind an unexpected (especially when we transform into just value like `.toMillis()`) so we might better to guard against that? 

Also thought about if we should break out the loop if the remaining duration becomes zero, but reconsidered as it would be better to give all units a chance to cleanup stuff even with zero timeout.